java-api-core/src/com/kinvey/java/store/BaseDataStore.java

```diff
@@ -315,6 +315,23 @@ public Integer count(KinveyCachedClientCallback<Integer> cachedCallback) throws
         return new ReadCountRequest<T>(cache, networkManager, this.storeType.readPolicy, null, client.getSyncManager()).execute().getCount();
     }
 
+    /**
+     * Get count of queried items in collection
+     * @param cachedCallback is using with StoreType.CACHE to get items count in collection
+     * @return count of queried items in collection
+     */
+    @Nonnull
+    public Integer count(KinveyCachedClientCallback<Integer> cachedCallback, Query query) throws IOException {
```

Comment:
Since we are deprecating the `.Cache` store type, I do not think that we need to have a method that takes a `cachedCallback` as a parameter.

Reply:
We have this logic for count request without query, and I think we need it for new case, because I guess we can delete `KinveyCachedClientCallback` logic only when we will delete `CACHE` store type from project, WDYT?

Reply:
This is a good point. But in this case, since this is new functionality, I believe that we can introduce this method without having the cache callback. If users want to use this method, they should use it with an actively-supported data store.